rossum_api/api_client.py

```diff
@@ -175,14 +177,20 @@ async def fetch_all(
         )
         # Fire async tasks to fetch the rest of the pages and start yielding results from page 1
         last_page = min(total_pages, max_pages or total_pages)
-        page_requests = [
-            asyncio.create_task(
-                self._fetch_page(
-                    f"{resource}", method, {**query_params, "page": i}, sideloads, json=json
+
+        in_flight_guard = asyncio.Semaphore(self.max_in_flight_requests)
+
+        async def _fetch_page(page_number):
+            async with in_flight_guard:
+                return await self._fetch_page(
+                    f"{resource}",
```

Comment:
`f"{resource}"` is the same as `resource` in this case. But I accept it was in the previous code. Up to you.